### PR TITLE
refactor(config): represent field resolver as separate enum

### PR DESCRIPTION
**Pull request by meskill**
**Summary:**  
Represent resolver for field as separate enum.

Pros of doing this:
- resolver now could be reused for types without copy-pasting
- single place to manage resolvers and avoid errors when adding new resolvers
- fixes present minor issues for resolvers check across the codebase

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

**Review by karatakis — APPROVED** [review]
Everything okay to me.

**Review by karatakis — APPROVED** [review]

File: src/core/config/config.rs [hunk 1]
```diff
@@ -209,6 +210,85 @@ pub struct RootSchema {
 /// Used to omit a field from public consumption.
 pub struct Omit {}
 
+// generate Resolver with macro in order to autogenerate conversion code
+// from the underlying directives.
+// TODO: replace with derive macro
```
Comment (tusharmath) [inline src/core/config/config.rs:215]: Create an issue for this.

**Comment by tusharmath** [comment]
The generated schema shouldn't have changed ideally.

**Comment by meskill** [comment]
> **The generated schema shouldn't have changed ideally.** Screenshot 2024-08-14 at 6 37 22 PM

This is a pure json schema issue. See https://github.com/GREsau/schemars/issues/317. The actual parsing and graphql schema work as expected.

File: src/core/config/config.rs (hunk 1 above)
Comment (meskill) [reply src/core/config/config.rs:215]: https://github.com/tailcallhq/tailcall/issues/2695

**Pull request merged**